### PR TITLE
create methods Services,Dao,Util

### PR DESCRIPTION
**Pull request by ConfusedSatlan**

### CODE REVIEW

File: src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java [hunk 1]
```diff
@@ -12,11 +15,37 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
...
+        Session session = factory.openSession();
+        try {
+            return Optional.ofNullable(session.get(Actor.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get actor from db with id: " + id, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```
Comment (IvanFadieiev) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java:49]: i think we might use TRW here

File: src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java [hunk 2]
```diff
@@ -12,11 +15,37 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
...
+        Session session = factory.openSession();
+        try {
+            return Optional.ofNullable(session.get(Country.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get actor from db with id: " + id, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```
Comment (IvanFadieiev) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java:49]: same here

File: src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -12,11 +15,37 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
...
+        Session session = factory.openSession();
+        try {
+            return Optional.ofNullable(session.get(Movie.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get actor from db with id: " + id, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```
Comment (IvanFadieiev) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java:49]: same here

**Review by IvanFadieiev — CHANGES_REQUESTED** [review]
Great solution! Some small details may be changed.

**Review by IvanFadieiev — APPROVED** [review]

File: src/main/java/mate/academy/hibernate/relations/Main.java [hunk 4]
```diff
@@ -16,17 +22,23 @@ public static void main(String[] args) {
         SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
 
         Country usa = new Country("USA");
-        CountryService countryService = null; // TODO: initialize this instance
+        CountryDaoImpl countryDao =
+                new CountryDaoImpl(HibernateUtil.getSessionFactory());
```
Comment (vovasalyha) [inline src/main/java/mate/academy/hibernate/relations/Main.java:26]: You have a variable `sessionFactory` on the line 22, so reuse it.

File: src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java [hunk 5]
```diff
@@ -12,11 +15,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
```
Comment (vovasalyha) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java:23]: It's better to use `persist` as you do not need any return value here, but method `save` returns something.

**Review by vovasalyha — APPROVED** [review]
Great work, approving, but take a look on my comments

**Pull request closed without merging**